docs/demos/coast-to-coast-q4-2023.md

```diff
@@ -0,0 +1,793 @@
+
+## Prerequisites
+The following prerequisites are required:
+- A running Kubernetes cluster where you have admin priveleges
+- `operator-sdk` installed. Installation instructions can be found at https://sdk.operatorframework.io/docs/installation/
+- `kubectl` installed. Installation instructions can be found at https://kubernetes.io/docs/tasks/tools/
+- a container runtime of your choice. In this demo we will be using `docker`
+- Install `operator-controller` vX.Y.Z. Installation instructions can be found at https://github.com/operator-framework/operator-controller/releases/tag/vX.Y.Z
+- `kustomize` installed. Installation instructions can be found at https://kubectl.docs.kubernetes.io/installation/kustomize/
+
+## Prepare for takeoff
+
+Before we can start exploring the new features we need to do some preparation by creating a new
+operator and building a few different versions.
+
+### Create an Operator
+>[!NOTE]
+> For the purposes of this demo we aren't actually going to make the operator do anything
+
+Using the `operator-sdk` we will create a new Operator.
+
+Create a new directory for the project and `cd` into it:
+```sh
+mkdir coastal && cd coastal
+```
+
+Initialize the project:
+```sh
+operator-sdk init --domain coastal.operatorframework.io
+```
+
+Create a new API:
+```sh
+operator-sdk create api \
+    --group coastal.operatorframework.io \
+    --version v1alpha1 \
+    --kind Coast \
+    --resource --controller
+```
+
+### Build the Operator Controller and Bundle Images
+For this demo we are going to build a couple different versions of this operator:
+- `v1.0.0-alpha1`
+- `v1.0.0`
+- `v1.0.1`
+- `v1.1.0`
+- `v2.0.0`
+
+Initial setup:
+- Create the `manifests/` directory for building the plain bundle
+```sh
+  mkdir -p manifests
+```
+
+- Create a Dockerfile for the plain bundle image
+```sh
+cat << EOF > plainbundle.Dockerfile
+FROM scratch
+ADD manifests /manifests
+EOF
+```
+
+For each of the versions above, do the following:
+- Generate the manifests
+```sh
+  make generate manifests
+```
+
+- Build the controller image
+```sh
+  make docker-build IMG="quay.io/operator-framework/coastal:{VERSION}"
+```
+
+- Populate the `manifests/` directory for building the plain bundle
+```sh
+  kustomize build config/default > manifests/manifests.yaml
+```
+
+- Build the plain bundle image
+```sh
+docker build -t quay.io/operator-framework/coastal-bundle:{VERSION} -f plainbundle.Dockerfile .
+```
+
+- Push the controller image and plain bundle image
+```sh
+docker push quay.io/operator-framework/coastal:{VERSION} && \
+docker push quay.io/operator-framework/coastal-bundle:{VERSION}
+```
+
+A handy little script to run all these steps for you:
+```sh
+#!/bin/bash
+set -e
+
+mkdir -p manifests
+
+cat << EOF > plainbundle.Dockerfile
+FROM scratch
+ADD manifests /manifests
+EOF
+
+versions=( v1.0.0-alpha1 v1.0.0 v1.0.1 v1.1.0 v2.0.0 )
+for version in "${versions[@]}"
```

Comment:
I personally prefer for readability (but I might be an outlier)

```suggestion
versions="v1.0.0-alpha1 v1.0.0 v1.0.1 v1.1.0 v2.0.0"
for version in "$versions"
```

Reply:
[9cecd82](https://github.com/operator-framework/operator-controller/pull/533/commits/9cecd82c89aa82af486c15b987cc97833517ffbf)

Reply:
I ended up reverting this as the loop only ran once with this change instead of iterating over all the versions